elasticsearch-akka/src/main/scala/com/sumologic/elasticsearch/akkahelpers/ScanAndScrollSource.scala

```diff
@@ -53,18 +53,23 @@ class ScanAndScrollSource(index: Index, tpe: Type, query: QueryRoot, scrollSourc
 
   val logger = LoggerFactory.getLogger(ScanAndScrollSource.getClass)
   override def preStart(): Unit = {
-    scrollSource.startScrollRequest(index, tpe, query).map { scrollResult =>
-      ScrollStarted(scrollResult)
+    scrollSource.startScrollRequest(index, tpe, query).map { case (scrollId, newData) =>
+      GotData(scrollId, newData)
     }.recover(recovery).pipeTo(self)
-    startWith(Starting, NotReady)
+    startWith(Starting, FirstScroll)
   }
 
   when(Starting) {
     case Event(ActorPublisherMessage.Request(_), _) => stay()
     case Event(ActorPublisherMessage.Cancel, _) => stop()
 
-    case Event(ScrollStarted(scrollId), NotReady) =>
-      requestMore(scrollId)
+    case Event(GotData(nextId, data), FirstScroll) =>
+      if (data.length == 0) {
+        onComplete()
+        stop()
+      } else {
+        consumeIfPossible(nextId, data)
+      }
```

Comment:
This is the same as the `GotData` case further down, so I'd vote for combining them.